api.go

```diff
@@ -143,7 +87,7 @@ func MakeWhitelist(
 // against a license whitelist, returning an error of all offending
 // DependencyLocks and their non-whitelisted licenses
 type Auditor interface {
-	Audit(...DependencyLock) error
+	Audit(...DependencyLock) ([]DependencyLock, error)
```

Comment:
Part of the reason I deleted the license locker code was that it wrote all of the identified licenses to the license.lock file _before_ auditing them to check whether they were on the whitelist or not. Meaning that even though the `adlrtool evaluate` command threw an error (returned from this Audit call) it would still produce a license file where some of the licenses are valid and some are not.

I updated Audit interface to return a list of verified licenses and with an error containing the list of invalid licenses.